### ISSUE TITLE
Can't push/pull to repo app after windows reinstall

### ISSUE THREAD

**Issue by luicaps** [heroku/heroku#1488]
Hey guys,

This is kinda weird, I had to format my pc last week, but I kept my files, and when I installed heroku CLI I couldn't do anything with my repo, git is always asking for credntials.

I logged in several times, tried using the API Key both on heroku and Git, but no use. I also cloned the repo again using heroku CLI, but again, I couldn't push to it, same reasons.

When I try to use any credentials on git, I get the usual: 
```
remote: !       WARNING:
remote: !       Do not authenticate with username and password using git.
remote: !       Run `heroku login` to update your credentials, then retry the git command.
remote: !       See documentation for details: https://devcenter.heroku.com/articles/http-git#authentication
fatal: Authentication failed for 'https://git.heroku.com/app.git/'
```

If I try using the "blank" user and API Key as credentials on git, I get:
```
To https://git.heroku.com/app.git
 ! [rejected]        master -> master (non-fast-forward)
error: failed to push some refs to 'https://git.heroku.com/app.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Trying my credentials to login on heroku both with username and password, or the API Key works fine, with an "Authentication successful." message.

I tried also using the "heroku git:remote" command on both my old repo and the new cloned one, and from both I get "Git remote heroku updated". I also used the name of my app with the command, and the update worked.

I've read the git auth article on dev center, but I found nothing based on what is happening to me, that is, git is always asking for credentials, and never connecting automatically. I saw some questions around the web with answers asking for a "~/.netrc" file, but I didn't understand how to use this information (there are some lots of files with this format name, on Heroku CLI folder).

Am I missing anything?

Thanks!

**Comment by dickeyxxx** [comment]
this happens because git and heroku aren't using the same _netrc file. This is something I'm trying to fix, but you can help me out by finding where your _netrc file should be and where the CLI is putting it.

Can you tell me which of the following paths has the _netrc file?

```
> echo %HOME%
> echo %HOMEDRIVE%%HOMEPATH%
> echo %USERPROFILE%
```

you could try to add the _netrc file into one of them to find the one that matches.

**Comment by luicaps** [comment]
I found the _netrc file on USERPROFILE (which is the same as HOMEDRIVE+HOMEPATH). I don't have the %HOME% var though.

The content of the file shows api.heroku.com and git.heroku.com login info

How can I find which _netrc is git using?

**Comment by dickeyxxx** [comment]
@luicaps the only way I know how to find out which one git is using is to add the `_netrc` to a folder you think might be a home directory and see if heroku does not prompt you for the token. There might be a better way.

**Comment by luicaps** [comment]
Ok, I found a trick. I created the HOME env var with the same value as the others (my home path, which is where the _netrc file is).

Does it mean that this is actually a "git issue" somehow?

**Comment by dickeyxxx** [comment]
it's not a git issue, adding a HOME env var will fix the problem for you, but I'd like heroku to use the same _netrc location as git for everything so other users don't have this issue

**Comment by dickeyxxx** [comment]
I would try to add it to the USERPROFILE location without a HOME env var and see if that fixes it

**Comment by luicaps** [comment]
Hope it works! Thanks!

**Comment by dickeyxxx** [comment]
another thing you could try is running `set` to see all the env vars

**Comment by luicaps** [comment]
Oh, I got now what you said! For me the USERPROFILE was actually right, pointing at the same _netrc file. My case was I didn't have HOME.

Somehow it was skipping the other vars

**Comment by dickeyxxx** [comment]
does anything else have the same directory as `USERPROFILE` if you run `set`? because that's the opposite behavior I have seen from most users. Also, what version of windows?

**Comment by luicaps** [comment]
No, the directory for USERPROFILE is only on the vars you mentioned (now that I created HOME too).

Do you want me to remove any var to see what happens (or change their values)?

I'm running on windows 8.

**Comment by dickeyxxx** [comment]
actually, I apologize. Reading through this message it seems that the only way it works for you is if you set the `HOME` var to something? Do you know where git is looking for the `_netrc` file if the `HOME` var isn't set?

**Comment by dickeyxxx** [comment]
you said `USERPROFILE` and `HOMESHARE+HOMEPATH` were the same. I'm curious where git was looking in your case without `HOME` being set.

**Comment by luicaps** [comment]
Yes, that's it. I removed the HOME var, and git started asking for credentials again.

How can I see where git is looking?

**Comment by dickeyxxx** [comment]
I don't know a good way to find out where git is looking aside from copying the `_netrc` file into a folder you think might be it and seeing if it stops asking for creds.

I would check the output of all of them with `set` and see if there are any paths in there you think might be it.

**Comment by dickeyxxx** [comment]
another question: if the HOME var isn't set, does the cli actually create the `_netrc` file? or does the HOME var need to be set for that to happen?

**Comment by S-Aoi** [comment]
I have the same issue. My `%home%` var isn't set. I can find `_netrc` file created in my `%userprofile%` folder after `heroku login`. I am on Windows 10.
I copy `_netrc` to `C:\`, `C:\Users`, `C:\Users\Public`, `C:\Windows`, `C:\Windows\System32`, `C:\Windows\SysWOW64` and `Git/usr`, but still can't push.

**Comment by Kosar79** [comment]
I created HOME variable equal to USERPROFILE and it works fine. 
> setx %HOME% %USERPROFILE%
